gdx/src/com/badlogic/gdx/graphics/g2d/PixmapPacker.java

```diff
@@ -326,66 +322,61 @@ public synchronized void dispose () {
-			if (page.texture == null) {
-				if (page.rects.size != 0 && page.addedRects.size > 0) {
-					page.texture = new Texture(new PixmapTextureData(page.image, page.image.getFormat(), useMipMaps, false, true)) {
-						@Override
-						public void dispose () {
-							super.dispose();
-							getTextureData().consumePixmap().dispose();
-						}
-					};
-					page.texture.setFilter(minFilter, magFilter);
-
-					for (String name : page.addedRects) {
-						Rectangle rect = page.rects.get(name);
-						TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
-							(int)rect.height);
-						atlas.addRegion(name, region);
-					}
-					page.addedRects.clear();
-				}
-			} else {
-				if (page.addedRects.size > 0) {
+			if (page.addedRects.size > 0) {
+				if (page.texture == null) {
+					generatePageTexture(page, minFilter, magFilter, useMipMaps);
+				} else {
 					page.texture.load(page.texture.getTextureData());
-					for (String name : page.addedRects) {
-						Rectangle rect = page.rects.get(name);
-						TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
+				}
+				for (String name : page.addedRects) {
+					Rectangle rect = page.rects.get(name);
+					TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
 							(int)rect.height);
-						atlas.addRegion(name, region);
-					}
-					page.addedRects.clear();
+					atlas.addRegion(name, region);
 				}
+				page.addedRects.clear();
+			}
+		}
+	}
+
+	private void generatePageTexture(final Page page, TextureFilter minFilter, TextureFilter magFilter, boolean useMipMaps) {
+		page.texture = new Texture(new PixmapTextureData(page.image, page.image.getFormat(), useMipMaps, false, true)) {
+			@Override
+			public void dispose() {
+				super.dispose();
+				getTextureData().consumePixmap().dispose();
+			}
+		};
+		page.texture.setFilter(minFilter, magFilter);
+	}
+
+	/** Return an ordered array of texture regions corresponding with the pages of the packer, creating new textures
+	 * if necessary.  The same array will be returned from successive calls to this method, possibly modified with
+	 * additional texture regions if more pages have been added since the last call.  Once this method has been called,
+	 * you must dispose of the resulting texture regions (or equivalently, any atlas generated or updated by this packer)
+	 * yourself - disposing the PixmapPacker will no longer dispose these textures or their backing pixmaps.
+	 */
+	public Array<TextureRegion> getTextureRegions(TextureFilter minFilter, TextureFilter magFilter, boolean useMipMaps) {
+		while (textureRegions.size < pages.size) {
+			if (disposed) return null;
+			Page page = pages.get(textureRegions.size);
+			if (page.texture == null) {
+				generatePageTexture(page, minFilter, magFilter, useMipMaps);
 			}
+			textureRegions.add(new TextureRegion(page.texture));
 		}
+		return textureRegions;

```

Comment:
After making this change locally and updating the tests, I'm actually inclining in a different direction now.  How would you feel if `updateTextureAtlas` did not clear `page.addedRects`, and also not call `texture.load()`.  After using `updateTextureRegions`, you need to `generateTextureAtlas` or `updateTextureAtlas` to get all the textures finalised (and the javadoc on `updateTextureRegions` would have to spell that out).

My reasons for this change of heart:
* The same gotcha we already discussed.  This came out as an extreme case when I converted my new FreeTypeAtlasTest - after generating the fonts, the atlas gets no TextureRegions (since all pages have `addedRects` empty) and therefore gets no Textures.
* You need to manually dispose something when you're done, and if you're passing your own packer to FreeTypeFontGenerator, the individual fonts can't own the textureRegion array (because it's potentially shared between multiple fonts).  A complete TextureAtlas generated from the PixmapPacker after finishing generating the fonts seems to me like the most natural thing to dispose.
* Efficiency.  If `updateTextureRegions` doesn't update the textures but `updateTextureAtlas` does, then packing lots of fonts using your custom packer incurs at most one additional `texture.load` per page.  The texture is created when the packer finishes packing the glyphs, and then updated when you update your TextureAtlas after generating all the fonts.  If `updateTextureRegions` did a `texture.load` for added rects, it would refresh the texture for each generated font (and if it didn't clear `addedRects` and didn't get some other way to determine that a texture needed refreshing, it would refresh all existing pages even if they hadn't changed since the last call.

OTOH, it feels a bit ugly for the new `updateTextureRegions` to give you back a Texture which might not be fully up-to-date and requires a subsequent call to `generateTextureAtlas` or `updateTextureAtlas` to finalise it, even if it is clearly flagged in the javadoc.  So, I'll wait and see what you think before proceeding.